### PR TITLE
[SYCL] Fix kernel shortcut path for inorder queue

### PR DESCRIPTION
**Pull request by KseniyaTikhomirova**
Commit https://github.com/intel/llvm/pull/11758 added extra event as dependency to fix host task vs kernel dependencies for inorder queue. That logic missed part when we try to bypass scheduler for kernel with no dependencies. This commit fixes it and allows to bypass scheduler if previous commands are enqueued (no host task is involved).

### CODE REVIEW

**Comment by KseniyaTikhomirova** [comment]
@intel/sycl-graphs-reviewers kindly ping

File: sycl/source/detail/queue_impl.hpp [hunk 1]
```diff
@@ -783,7 +783,6 @@ class queue_impl {
       EventRet = Handler.finalize();
   }
 
-protected:
```
Comment (julianmi) [inline sycl/source/detail/queue_impl.hpp:786]: Why is this removed?

Comment (KseniyaTikhomirova) [reply sycl/source/detail/queue_impl.hpp:786]: it is a duplication. protected title is already present on l748

Comment (julianmi) [reply sycl/source/detail/queue_impl.hpp:786]: Thanks for clarifying.

**Review by julianmi — APPROVED** [review]
